sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -76,6 +78,15 @@
 CONTENT_CHUNK_SIZE = 10 * 1024
 _LOGGER = logging.getLogger(__name__)
 
+try:
+    # ConnectionTimeoutError was only introduced in aiohttp 3.10 so we want to keep this
+    # backwards compatible. If client is using aiohttp <3.10, the behaviour will safely
+    # fall back to treating a TimeoutError as a ServiceResponseError (that wont be retried).
+    from aiohttp.client_exceptions import ConnectionTimeoutError
+except ImportError:
+
+    class ConnectionTimeoutError(asyncio.TimeoutError): ...  # type: ignore[no-redef]
```

Comment:
```suggestion
    class ConnectionTimeoutError(Exception): ...  # type: ignore[no-redef]
```

Reply:
This should allow you to properly catch ConnectionTimeoutError without risking stomping on the old behavior below, no?

Reply:
Done!